### PR TITLE
Update DG list edit

### PR DESCRIPTION
**Pull request by kohkaixun**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -325,6 +325,46 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    Use case ends
+
+**Extensions**
+
+* 1a. InternBuddy detects that index of internship entry given is not a number.
+  * 1a1. InternBuddy prompts user for a valid internship entry index.
+  
+    Use case resumes from Step (1).
+
+* 1b. InternBuddy detects that index of internship given is out of range.
```
Comment (potty10) [inline docs/DeveloperGuide.md:344]: I feel that it is a bit vague to say out of range. Is it better to write "an internship entry is not found using the given user-supplied argument"?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -325,6 +325,46 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**MSS**
+
+1.  User edits an internship entry.
+2.  InternBuddy updates that particular internship entry.
+
+    Use case ends
+
+**Extensions**
+
+* 1a. InternBuddy detects that index of internship entry given is not a number.
```
Comment (potty10) [inline docs/DeveloperGuide.md:339]: Same as line 344.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -325,6 +325,46 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
       Use case ends.
 
+**Use Case: Edit an internship entry.**
+
+**MSS**
+
+1.  User edits an internship entry.
+2.  InternBuddy updates that particular internship entry.
```
Comment (potty10) [inline docs/DeveloperGuide.md:333]: Is it also better to say "and displays success message"?

Comment (kohkaixun) [reply docs/DeveloperGuide.md]: Ok noted, will add the "display success message" in.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (kohkaixun) [reply docs/DeveloperGuide.md]: Ok after discussion, will merge 1a and 1b into 1c and 1d since they have much overlap.

**Comment by potty10** [comment]
LGTM!

**Pull request merged**